.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,23 @@
+## Pull Request
+
+### All Submissions:
+
+- [ ] Have you followed the guidelines mentioned in our [Contributing document](/CONTRIBUTING.md)?
+- [ ] Have you ensured that there aren't other open [Pull Requests](../../../pulls) for the same update/change?
+- [ ] Is your [Pull Request](../../../pulls) linked to an [Issue](../../../issues)? If yes, please mention the Issue Number.
```

Comment:
```suggestion
### Checklist:
- [ ] Followed the guidelines mentioned in our [Contributing document](/CONTRIBUTING.md)
- [ ] Ensured that there aren't other open [Pull Requests](../../../pulls) for the same update/change
- [ ] This PR is linked to an [Issue](../../../issues) and the issue number has been mentioned in the title and body of the PR 
- [ ] The code is clean and docstrings have been added or updated as required
- [ ] Code is linted/formatted locally prior to submission (using `black`)
```

<!-- Please ensure you get maximum checks done before submitting a PR, for quick review and merge -->